javascript/javascript3/week1/weather-app.js

```diff
@@ -0,0 +1,110 @@
+//Declaring the variables and getting the DOM elements
+const userInput = document.getElementById('cityName');
+const submitButton = document.getElementById('submit');
+const warningMsg = document.getElementById('warning-msg');
+const getCurrentPosition = document.getElementById('location-button');
+const displayWeather = document.getElementById('display-weather');
+const cityName = document.getElementById('city-name');
+const temperature = document.getElementById('temperature');
+const weatherIcon = document.getElementById('weather-icon');
+const imageIcon = document.createElement('img');
+const wind = document.getElementById('wind');
+const cloudy = document.getElementById('cloudy');
+const sunrise = document.getElementById('sunrise');
+const sunset = document.getElementById('sunset');
+let lat;
+let lng;
+let setLocalStorageForButton1;
+let setLocalStorageForButton2;
+
+submitButton.addEventListener('click', (event) => {
+    const cityName = userInput.value;
+    console.log(cityName)
+    if (cityName === '') {
+        warningMsg.innerHTML = 'Please enter a city name';
+    }
+    else {
+        warningMsg.innerHTML = '';
+        //Api to fetch weather information using user entered city 
+        fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=4885c4b997291cbbdb529b59964ffed7`)
+            .then(response => response.json())
+            .then(weatherData => {
+                displayWeatherDetails(weatherData);
+                setLocalStorageForButton1 = cityName.innerHTML;
+                localStorage.setItem('userCity', setLocalStorageForButton1);
+            });
+    }
+});
+
+//to get the position of user
+getCurrentPosition.addEventListener('click', (event) => {
+    if (!navigator.geolocation) {
+        status.textContent = 'Geolocation is not supported by your browser';
+    }
+    else {
+        status.textContent = 'Locating…';
+        navigator.geolocation.getCurrentPosition(success, error);
+    }
+    function error() {
+        status.textContent = 'Unable to retrieve your location';
+    }
+    function success(position) {
+        const latitude = position.coords.latitude;
+        const longitude = position.coords.longitude;
+        //Api to fetch weather information using position
+        fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=4885c4b997291cbbdb529b59964ffed7`)
+            .then(response => response.json())
+            .then(weatherData => {
+                displayWeatherDetails(weatherData);
+                setLocalStorageForButton2 = cityName.innerHTML;
+                localStorage.setItem('localWeather', setLocalStorageForButton2);
+            });
+    }
+});
+
+let localStorageData;
```

Comment:
Would be more comprehensive if the global variables define at the top of the codes. Maybe after line 18.